homeassistant/components/vizio/media_player.py

```diff
@@ -155,18 +159,26 @@ def __init__(
             CONF_ADDITIONAL_CONFIGS, []
         )
         self._device = device
-        self._max_volume = float(self._device.get_max_volume())
+        self._max_volume = float(device.get_max_volume())
 
         # Entity class attributes that will change with each update (we only include
         # the ones that are initialized differently from the defaults)
         self._attr_sound_mode_list = []
         self._attr_supported_features = SUPPORTED_COMMANDS[device_class]
 
         # Entity class attributes that will not change
-        self._attr_name = name
         self._attr_icon = ICON[device_class]
-        self._attr_unique_id = self._config_entry.unique_id
+        unique_id = config_entry.unique_id
+        assert unique_id
```

Comment:
why is this assertion needed?

Reply:
Because config_entry.unique_id can technically be None

Reply:
Not for this PR, but maybe we should introduce a class `ConfigEntryWithUniqueId` or something to fix this? Or make ConfigEntry a generic class, where the unique ID type can be specified?